implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OtlpConfigTest.java

```diff
@@ -33,4 +40,40 @@ void resourceAttributesInputParsing() {
         assertThat(config.resourceAttributes()).containsEntry("k", "v").containsEntry("a", "b").hasSize(2);
     }
 
+    @Test
+    void headersEmptyishInputParsing() {
+        Stream<OtlpConfig> configs = Stream.of(k -> null, k -> "", k -> "  ", k -> " ,");
+        configs.forEach(config -> assertThat(config.headers()).isEmpty());
+    }
+
+    @Test
+    void headersConfigTakesPrecedenceOverEnvVars() throws Exception {
+        OtlpConfig config = k -> "header1=value1";
+        withEnvironmentVariable("OTEL_EXPORTER_OTLP_HEADERS", "header2=value")
```

Comment:
I found the [system-stubs](https://github.com/webcompere/system-stubs) library, which allows us to test with different environment variables. We should update and add tests for the `resourceAttributes` config - it also depends on environment variables.